ui/ozone/platform/x11/x11_display_manager_ozone.cc

```diff
@@ -23,34 +24,36 @@ namespace {
 
 constexpr int default_refresh = 60;
 
-X11DisplayManagerOzone::Snapshot CreateSnapshot(int64_t display_id,
-                                                gfx::Rect bounds,
-                                                gfx::ColorSpace color_space) {
-  std::unique_ptr<display::DisplaySnapshot> snapshot =
-      std::make_unique<display::DisplaySnapshot>(
-          display_id, gfx::Point(bounds.x(), bounds.y()),
-          gfx::Size(bounds.width(), bounds.height()),
-          display::DisplayConnectionType::DISPLAY_CONNECTION_TYPE_NONE, false,
-          false, false, color_space, "", base::FilePath(),
-          display::DisplaySnapshot::DisplayModeList(), std::vector<uint8_t>(),
-          nullptr, nullptr, 0, 0, gfx::Size());
-  std::unique_ptr<display::DisplayMode> current_mode =
+std::unique_ptr<display::DisplaySnapshot> CreateSnapshot(
+    int64_t display_id,
+    gfx::Rect bounds,
+    gfx::ColorSpace color_space) {
+  display::DisplaySnapshot::DisplayModeList modes;
+  std::unique_ptr<display::DisplayMode> display_mode =
       std::make_unique<display::DisplayMode>(
           gfx::Size(bounds.width(), bounds.height()), false, default_refresh);
-  snapshot->set_current_mode(current_mode.get());
-  return std::make_pair(std::move(snapshot), std::move(current_mode));
+  modes.push_back(std::move(display_mode));
+  const display::DisplayMode* mode = modes.back().get();
+
+  return std::make_unique<display::DisplaySnapshot>(
+      display_id, gfx::Point(bounds.x(), bounds.y()),
+      gfx::Size(bounds.width(), bounds.height()),
+      display::DisplayConnectionType::DISPLAY_CONNECTION_TYPE_NONE, false,
+      false, false, color_space, "", base::FilePath(), std::move(modes),
+      std::vector<uint8_t>(), mode, mode, 0, 0, gfx::Size());
 }
 
 void BuildFallbackDisplayList(
-    X11DisplayManagerOzone::OwnedDisplaySnapshot& output) {
+    std::vector<std::unique_ptr<display::DisplaySnapshot>>& output) {
```

Comment:
AFAIK, current agreement is to pass parameters by pointers if they are going to be changed. Please, check chromium-style guide. Or you can change the return type and build this vector here and return it.